### PR TITLE
Start using gem instead of obsolete library/tool for Nessus bridge

### PR DESCRIPTION
**Pull request by kost**
Rationale is following:
nessus-cli is obsolete
nessus is using json rest api instead of xmlrpc
xmlrpc name is therefore obsolete

Solution: with minimal changes start using nessus_rest gem.

### CODE REVIEW

**Comment by void-in** [comment]
We would want to keep the XML RPC module as well. Please have a look at [this ](https://github.com/rapid7/metasploit-framework/pull/5112#issuecomment-91369794)comment from @wchen-r7

**Comment by kost** [comment]
There's no problem with auxiliary module/scanners as they use their own implementation of protocol(s).
I.e. if we're are talking about nessus_xmlrpc_ping and nessus_xmlrpc_login, I can confirm that as I'm the sole author of them :)
(you can also check requires in coresponding .rb files, even in nessus_rest_login.rb of which I'm not author)

Regarding keeping Nessus XMLRPC support inside MSF bridge(s), i like to compare vulnerability scanners with anti-virus software - what's the purpose of scanning for malware with 2 or 3 year old anti-virus? :)

**Comment by bcook-r7** [comment]
Good point @kost , this makes sense to me, especially for the bridge plugin (as opposed to modules, which is a different matter.)

File: plugins/nessus.rb [hunk 1]
```diff
@@ -1065,11 +1075,17 @@ def cmd_nessus_db_scan(*args)
...
+        et['launch']='ONETIME'
+        et['name']=name
+        et['text_targets']=targets
+        et['description']=desc
+        et['launch_now']=true
+        scan = @n.scan_create(policy_id, et)
         if !scan["error"]
           scan = scan["scan"]
-          print_status("Scan ID #{scan['id']} successfully created")
-          print_status("Run nessus_scan_launch #{scan['id']} to launch the scan")
```
Comment (egypt) [inline plugins/nessus.rb]: Is the `nessus_scan_launch` command still needed after this change?

Comment (kost) [inline plugins/nessus.rb]: Yes, if you create the scan with nessus_scan_new... and also if you create the scan from Nessus UI and want to launch it from the mf.

**Comment by kost** [comment]
Yes, if you create the scan with nessus_scan_new... and also if you create the scan from Nessus UI and want to launch it from the msf.

**Comment by pbarry-r7** [comment]
I verified these changes worked fine for me against a Nessus server instance.  Thanks for this update, @kost!

**Comment by pbarry-r7** [comment]
# Release Notes

This patch shifts Nessus bridge plugin support in framework from older, built-in code to the newer, more up-to-date nessus_rest gem.

**Pull request merged**